json/tracing_test.go

```diff
@@ -0,0 +1,68 @@
+package json_test
+
+import (
+	"github.com/stretchr/testify/require"
+	"github.com/uber/tchannel-go"
+	"github.com/uber/tchannel-go/json"
+	. "github.com/uber/tchannel-go/testutils/testtracing"
+	"golang.org/x/net/context"
+	"testing"
+)
+
+// JSONHandler tests tracing over JSON encoding
+type JSONHandler struct {
+	TraceHandler
+	t *testing.T
+}
+
+func (h *JSONHandler) firstCall(ctx context.Context, req *TracingRequest) (*TracingResponse, error) {
+	jctx := json.Wrap(ctx)
+	var response TracingResponse
+	peer := h.Ch.Peers().GetOrAdd(h.Ch.PeerInfo().HostPort)
+	err := json.CallPeer(jctx, peer, h.Ch.PeerInfo().ServiceName, "call", req, &response)
+	require.NoError(h.t, err)
+	return &response, err
+}
+
+func (h *JSONHandler) callJSON(ctx json.Context, req *TracingRequest) (*TracingResponse, error) {
+	return h.HandleCall(ctx, req,
+		func(ctx context.Context, req *TracingRequest) (*TracingResponse, error) {
+			jctx := ctx.(json.Context)
+			sc := h.Ch.Peers().GetOrAdd(h.Ch.PeerInfo().HostPort)
+			childResp := new(TracingResponse)
```

Comment:
nit: to be consistent with above, let's do `var response TracingResponse`, or change the above to also be `response := new(TracingResponse)`